apps/teams-test-app/src/components/utils/ApiWithCheckboxInput.tsx

```diff
@@ -2,12 +2,18 @@ import * as React from 'react';
 
 import { noHostSdkMsg } from '../../App';
 import { ApiContainer } from './ApiContainer';
+import { getTestBackCompat } from './getTestBackCompat';
 
 export interface ApiWithCheckboxInputProps {
   title: string;
   name: string; // system identifiable unique name in context of Teams Client and should contain no spaces
   label: string;
-  onClick: (input: boolean) => Promise<string>;
+  onClick:
+    | ((input: boolean) => Promise<string>)
+    | {
+        withTeamsV2: (input: boolean) => Promise<string>;
```

Comment:
For the other ones we're using `withPromise` and `withCallback`. We should be consistent.

Reply:
Thanks. The Apis with checkboxes section are not updating from callbacks to promises, so I would think the `withPromise` and `withCallback` do not explain the scenarios here. If we use `withCallback` just to represent the compatibility layer, that would make sense. I will update them.

Reply:
Looking through this, I think we can still follow the `withCallback` and `withPromise` scenarios, the `withPromise` scenario will just replace `onClick.withTeamsV2`